src/SDKs/Azure.Base/data-plane/Azure.Base/Http/Pipeline/FixedRetryPolicy.cs

```diff
@@ -0,0 +1,51 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+
+namespace Azure.Base.Http.Pipeline
+{
+    internal class FixedRetryPolicy : RetryPolicy
+    {
+        private readonly Func<Exception, bool> _exceptionFilter;
+
+        private readonly int _maxRetries;
+
+        private readonly TimeSpan _delay;
+
+        private readonly int[] _retriableCodes;
+
+        public FixedRetryPolicy(int[] retriableCodes, Func<Exception, bool> exceptionFilter, int maxRetries, TimeSpan delay)
+        {
+            if (retriableCodes == null)
+            {
+                throw new ArgumentNullException(nameof(retriableCodes));
+            }
+
+            _exceptionFilter = exceptionFilter;
+            _maxRetries = maxRetries;
+            _delay = delay;
+
+            _retriableCodes = retriableCodes.ToArray();
+            Array.Sort(_retriableCodes);
+        }
+
+        protected override bool ShouldRetry(HttpPipelineMessage message, Exception exception, int attempted, out TimeSpan delay)
```

Comment:
Could we just receive the status instead of the whole message?

Reply:
Having the message allows us to handle things like `x-retry-after` in the future.

Reply:
I see. The way you have it right now means that the `maxRetries` and `delay` is set per client.
In the future, you will make changes so that at least the delay could be modified according to what the response from the server is?

Reply:
`ShouldRetry` method returns `delay` via the `out` parameter. In the current implementation, it always returns pre-configured value but you can imagine an implementation where `delay` parameter would be set based on, for example, `x-retry-after` header

Reply:
Yeah, we are saying the same thing. Cool